modules/host/announce.go

```diff
@@ -57,14 +65,22 @@ func (h *Host) Announce() error {
 	h.mu.RUnlock()
 
 	// Check that the host's ip address is known.
-	if addr.IsLoopback() {
+	if addr.IsLoopback() && build.Release != "testing" {
 		return errors.New("can't announce without knowing external IP")
 	}
 
 	return h.announce(addr)
 }
 
-// ForceAnnounce announces using the provided address.
+// AnnounceAddress submits a host announcement to the blockchain to announce a
+// specific address. No checks for validity are performed on the address.
 func (h *Host) AnnounceAddress(addr modules.NetAddress) error {
+	// As an exported function that needs resources, AnnouceAddress holds the
+	// resource lock throughout its operation.
```

Comment:
wrote it a few times, but really I should talk about it in the conventions documentation and then remove it from here.

Sounds like we haven't finished determining exactly what this usage should look like, so I'll hold off on writing more conventions stuff